pkgs/development/compilers/flutter/default.nix

```
@@ -31,19 +42,33 @@ in
     pname = "flutter";
     version = "3.3.3";
     dartVersion = "2.18.2";
-    hash = "sha256-MTZeWQUp4/TcPzYIT6eqIKSPUPvn2Mp/thOQzNgpTXg=";
+    flutterHash = {
+      linux = "sha256-MTZeWQUp4/TcPzYIT6eqIKSPUPvn2Mp/thOQzNgpTXg=";
+      aarch64-darwin = "sha256-apU+jgJgNZBAIffLjh6e6B5szwQsjjve3aVXZDPjD94=";
+    };
     dartHash = {
       x86_64-linux = "sha256-C3+YjecXLvSmJrLwi9H7TgD9Np0AArRWx3EdBrfQpTU";
       aarch64-linux = "sha256-zyIK1i5/9P2C+sjzdArhFwpVO4P+It+/X50l+n9gekI=";
+      aarch64-darwin = "sha256-Sf7WGsQyRAsoF2ZsjAkEDPWGpmU21N3Z6WPfvDCyn+M=";
     };
-    patches = getPatches ./patches/flutter3;
+    patches = [
+      ./patches/flutter3/disable-auto-update.patch
+      ./patches/flutter3/git-dir.patch
+      (if stdenv.hostPlatform.isLinux then
+        ./patches/flutter3/move-cache-linux.patch
+      else
+        ./patches/flutter3/move-cache-darwin.patch
+      )
+    ];
   };
 
   v2 = flutterDrv {
     pname = "flutter";
     version = "2.10.5";
     dartVersion = "2.16.2";
-    hash = "sha256-DTZwxlMUYk8NS1SaWUJolXjD+JnRW73Ps5CdRHDGnt0=";
+    flutterHash = {
+      linux = "sha256-DTZwxlMUYk8NS1SaWUJolXjD+JnRW73Ps5CdRHDGnt0=";
+    };
```

```suggestion
    flutterHash = {
      x86_64-linux = "sha256-DTZwxlMUYk8NS1SaWUJolXjD+JnRW73Ps5CdRHDGnt0=";
    };
```

---

It is hash for for arch-independent source of flutter. I think we need to leave at as linux.

---

We should individually list all arches here them. Having two different formats is really confusing.

---

If you don't want to repeat the hashes, you can do it like this:

```suggestion
    flutterHash = rec {
      x86_64-linux = "sha256-DTZwxlMUYk8NS1SaWUJolXjD+JnRW73Ps5CdRHDGnt0=";
      aarch64-linux = x86_64-linux;
    };
```